database/src/types.rs

```diff
@@ -49,96 +45,290 @@ use bytemuck::{AnyBitPattern, NoUninit, Pod, Zeroable};
+/// assert_eq!(&a, c);
+/// assert_eq!(c, c2.as_ref());
+/// ```
 ///
+/// # Size & Alignment
+/// ```rust
+/// # use cuprate_database::types::*;
+/// # use std::mem::*;
+/// assert_eq!(size_of::<BlockInfoV1>(), 64);
+/// assert_eq!(align_of::<BlockInfoV1>(), 8);
+/// ```
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[derive(Copy, Clone, Debug, PartialEq, PartialOrd, Eq, Ord, Hash, Pod, Zeroable)]
+#[repr(C)]
+pub struct BlockInfoV1 {
+    /// TODO
+    pub timestamp: u64,
+    /// TODO
+    pub total_generated_coins: u64,
+    /// TODO
+    pub weight: u64,
+    /// TODO
+    pub cumulative_difficulty: u64,
+    /// TODO
+    pub block_hash: [u8; 32],
+}
+
+//---------------------------------------------------------------------------------------------------- BlockInfoV2
+/// TODO
+///
+/// ```rust
+/// # use std::borrow::*;
+/// # use cuprate_database::{*, types::*};
 /// // Assert Storable is correct.
-/// let b2 = Storable::as_bytes(&a);
-/// let c2: &TestType = Storable::from_bytes(b2);
-/// assert_eq!(a, *c2);
-/// assert_eq!(b, b2);
-/// assert_eq!(c, *c2);
-/// assert_eq!(c2.u, 1);
-/// assert_eq!(c2.b, 255);
-/// assert_eq!(c2._pad, [0; 7]);
+/// let a = BlockInfoV2 {
+///     timestamp: 1,
+///     total_generated_coins: 123,
+///     weight: 321,
+///     cumulative_difficulty: 111,
+///     block_hash: [54; 32],
+///     cumulative_rct_outs: 2389,
+///     _pad: [7; 4],
+/// };
+/// let b = Storable::as_bytes(&a);
+/// let c: &BlockInfoV2 = Storable::from_bytes(b);
+/// let c2: Cow<'_, BlockInfoV2> = Storable::from_bytes_unaligned(b);
+/// assert_eq!(&a, c);
+/// assert_eq!(c, c2.as_ref());
 /// ```
 ///
 /// # Size & Alignment
 /// ```rust
 /// # use cuprate_database::types::*;
 /// # use std::mem::*;
-/// assert_eq!(size_of::<TestType>(), 16);
-/// assert_eq!(align_of::<TestType>(), 8);
+/// assert_eq!(size_of::<BlockInfoV2>(), 72);
+/// assert_eq!(align_of::<BlockInfoV2>(), 8);
 /// ```
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
 #[derive(Copy, Clone, Debug, PartialEq, PartialOrd, Eq, Ord, Hash, Pod, Zeroable)]
 #[repr(C)]
-pub struct TestType {
-    /// TEST
-    pub u: usize,
-    /// TEST
-    pub b: u8,
-    /// TEST
-    ///
-    /// TODO: is there a cheaper way (CPU instruction wise)
-    /// to add padding to structs over 0 filled arrays?
-    ///
-    /// TODO: this is basically leeway to
-    /// add more things to our structs too,
-    /// because otherwise this space is wasted.
-    pub _pad: [u8; 7],
+pub struct BlockInfoV2 {
+    /// TODO
+    pub timestamp: u64,
+    /// TODO
+    pub total_generated_coins: u64,
+    /// TODO
+    pub weight: u64,
+    /// TODO
+    pub cumulative_difficulty: u64,
+    /// TODO
+    pub block_hash: [u8; 32],
+    /// TODO
+    pub cumulative_rct_outs: u32,
+    /// TODO
+    // TODO: get rid of or use this padding.
+    // Without this, `BlockInfoV2` is 68 bytes,
+    // which isn't aligned to 8.
+    pub _pad: [u8; 4],
```

Comment:
We have 4 bytes to play with here.

Could do `cumulative_rct_outs: u64` if we don't want explicit padding.

Reply:
yeah I think just do `cumulative_rct_outs: u64`